x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/UpdateRolloverLifecycleDateStep.java

```diff
@@ -76,6 +72,24 @@ public ClusterState performAction(Index index, ClusterState currentState) {
             .put(newIndexMetadata)).build();
     }
 
+    private String getRolloverTarget(Index index, ClusterState currentState) {
+        IndexAbstraction indexAbstraction = currentState.metadata().getIndicesLookup().get(index.getName());
+        final String rolloverTarget;
+        if (indexAbstraction.getParentDataStream() != null) {
+            rolloverTarget = indexAbstraction.getParentDataStream().getDataStream().getName();
```

Comment:
Same comment here about `null` check for the `getDataStream()`, I also wonder if maybe we should make this a nice static helper like `IndexAbstraction.streamNameOrNull(indexAbstraction)`?

Reply:
The underlying `dataStream` is non-nullable though. The `getName` method that Dan recommended emphasis this so replaced it here as well. Happy to discuss if there are any situations where this could be null though (in which case I believe more null checks are needed in `IndexAbstraction.DataStream`)